rust/chains/hyperlane-ethereum/src/validator_announce.rs

```diff
@@ -110,6 +131,41 @@ where
             .await?;
         Ok(storage_locations)
     }
+
+    async fn announce_tokens_needed(
+        &self,
+        announcement: SignedType<Announcement>,
+    ) -> ChainResult<U256> {
+        let contract_call = self
+            .announce_contract_call(announcement.clone(), None)
+            .await?;
+        let validator = announcement.value.validator;
```

Comment:
Re-order to avoid a clone (You might need `let validator = announcement.value.validator.clone();`)
```suggestion
        let validator = announcement.value.validator;
        let contract_call = self
            .announce_contract_call(announcement, None)
            .await?;
```